api/types.proto

```diff
@@ -696,3 +696,9 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+message Device {
+	string source = 1;
+	string target = 2;
+	string permissions = 3;
```

Comment:
> For example, do we need to be able to say "mount anything that starts with /dev/snd" 

Just a quick remark: `/dev/snd` is not a prefix for multiple devices, `/dev/snd` is a folder containing other devices. i.e., it's `/dev/snd/controlC0` and not `/dev/snd0`.
For NVIDIA GPUs it's different because they indeed all have a `/dev/nvidia` prefix, but you actually want fine-grain control on which devices you import. If you want to use 1 GPU in your app you will only select `/dev/nvidia0`, if you want 2 GPUs you will select `/dev/nvidia0` and `/dev/nvidia1`. However, this consideration starts overlapping another discussion: GPU as a fungible resource in Swarm (like CPU/mem).

Thanks for the explanation, give me a little time to process this.